encoding/protoavro/marshal.go

```diff
@@ -10,10 +10,42 @@ import (
 	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
+type MarshalOptions struct {
+	ExtraFields []ExtraField
+}
+
+type ExtraField struct {
+	FieldName string
+	Message proto.Message
+}
+
+func (o MarshalOptions) convertToSchemaOptions() SchemaOptions {
+	extraFields := make([]SchemaExtraField, len(o.ExtraFields))
+	for i, field := range o.ExtraFields {
+		extraFields[i] = SchemaExtraField{
+			FieldName: field.FieldName,
+			Descriptor: field.Message.ProtoReflect().Descriptor(),
+		}
+	}
+	return SchemaOptions{ExtraFields: extraFields}
+}
+
+func (o MarshalOptions) ConvertToUnmarshalOptions() *UnmarshalOptions {
+	return &UnmarshalOptions{
+		ExtraFields: o.ExtraFields,
+		TrashCan: make([]proto.Message, 0),
```

Comment:
what's the rationale behind this name?

Reply:
Short answer: None. 
A bit longer: remnant of frustration trying to unmarshal extra fields to a slice to save somewhere.

Reply:
It seems to me like we're trying to shoehorn a solution into the wrong place here, IMO. Can we discuss if this bigger problem we're trying to solve can be more easily solved somewhere else?

Reply:
Let's set up some time?